chore: sync alpha to master

PKulkoRaccoonGang opened the pull request:
## Description

- Updated alpha branch
- added new tokens
- fixed build

## Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [ ] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [ ] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [ ] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [ ] Were your changes tested in the `example` app?
* [ ] Is there adequate test coverage for your changes?
* [ ] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please request an a11y review for the PR in the [#wg-paragon](https://openedx.slack.com/archives/C02NR285KV4) Open edX Slack channel.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@openedx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

github-actions[bot] committed:
docs: update dependent-usage.json (#2973)

khudym committed:
chore: remove browserlist-config (#2970)

* build: remove browserlist-config

* test: update snapshot

---------

Co-authored-by: Kyrylo Hudym-Levkovych <kyr.hudym@kyrs-MacBook-Pro.local>

khudym committed:
fix: center align Hyperlink with its icon (#2971)

Co-authored-by: Kyrylo Hudym-Levkovych <kyr.hudym@kyrs-MacBook-Pro.local>

khudym committed:
fix: update `TableFooter` to match design spec (#2969)

* fix: remove unnecessary margin

* fix: change data-table-border value

---------

Co-authored-by: Kyrylo Hudym-Levkovych <kyr.hudym@kyrs-MacBook-Pro.local>

edx-requirements-bot committed:
chore: update browserslist DB (#2967)

Co-authored-by: adamstankiewicz <adamstankiewicz@users.noreply.github.com>

github-actions[bot] committed:
docs: update dependent-usage.json (#2978)

github-actions[bot] committed:
docs: update dependent-usage.json (#2980)

PKulkoRaccoonGang committed:
chore: added Husky configuration (#2949)

Co-authored-by: Viktor Rusakov <52399399+viktorrusakov@users.noreply.github.com>

PKulkoRaccoonGang committed:
docs: switch scope of the package to `openedx` in docs (#2979)

* refactor: changed Paragon NPM package name and source

* refactor: example refactoring

* refactor: corrected deps usage analyzer

* refactor: defactoring deps usage analizer

* refactor: removed dependent-usage-analyzer from eslintignore

PKulkoRaccoonGang committed:
feat: added CLI command to migrate to using `openedx` scope (#2984)

* feat: added migration command to new openedx-paragon npm package

* chore: add default path to the cli

---------

Co-authored-by: Viktor Rusakov <vrusakov66@gmail.com>

adamstankiewicz committed:
docs: include Algolia DocSearch ADR and crawler config (#2991)

adamstankiewicz committed:
fix: rely on Form.Control and controlId in Form.Group (#2931)

brian-smith-tcril committed:
build: specify semantic release version that is compatible with node 18 (#2992)

semantic-release [version 23](https://github.com/semantic-release/semantic-release/releases/tag/v23.0.0) includes the following breaking change:
* support for node v18 has been dropped and the minimum for v20 is now v20.8.1

This pins the release workflow to version 22 until we move away from node 18.

adamstankiewicz committed:
feat!: release chip, searchfield, pagination, form.autosuggest (#2964)

adamstankiewicz committed:
Revert "feat!: release chip, searchfield, pagination, form.autosuggest (#2964)" (#2994)

This reverts commit 0274f10bc716bc78f1a306b0af4f2104a3b9abe3.

adamstankiewicz committed:
feat!: release updates to Chip, SearchField, Pagination, Form.Autosuggest (#2995)

BREAKING CHANGE: Many of the SCSS variables (i.e., tokens) surrounding `Chip` were removed. Consumers should verify no longer using any of the removed SCSS variables in custom Paragon brands/themes.

BREAKING CHANGE: Many of the SCSS variables (i.e., tokens) surrounding `Pagination` were removed. Consumers should verify no longer using any of the removed SCSS variables in custom Paragon brands/themes.

BREAKING CHANGE: `icons` prop on `SearchField` now accepts the icon src instead of an `Icon` component.

BREAKING CHANGE: `icons` prop on `Pagination` now accepts the icon src instead of an `Icon` component.

BREAKING CHANGE: `value` prop of `Form.Autosuggest` is now an object instead of a string

BREAKING CHANGE: `Form.Autosuggest` now uses `onChange` instead of `onSelected`

BREAKING CHANGE: `Form.Autosuggest` now takes in different error messages for value/selection required, and custom errors

adamstankiewicz committed:
build: temporarily disable cron on failing usage insights github actions (#3001)

brian-smith-tcril committed:
feat: allow consumers to configure ModalDialog to allow overflow (#2939)

syedsajjadkazmii committed:
fix: unrecognized prop warning in form autosuggest (#3003)

viktorrusakov committed:
build: ignore npm scripts when installing dependencies in dependent analyzer workflow (#3013)

* build: ignore npm scripts when installing dependencies in dependent analyzer workflow

* chore: upgrade husky to v9

bradenmacdonald committed:
feat: working typings for Paragon, better types for <Icon> component (#3016)

* feat: better types for <Icon> component

* fix: TypeScript rootDir now that it's checking '../src' files too

* chore: fix eslint 'import/order' & 'import/no-unresolved' issues in www

* fix: build wasn't including types properly

* fix: <Icon/> types still weren't correct

* fix: explicitly define other exports as having 'any' type

* fix: changing www/tsconfig.json is no longer needed

* fix: warning seen when gatsby parses index.d.ts during 'npm run start'

bradenmacdonald committed:
fix: display of menu items with checkboxes (#3019)

brian-smith-tcril committed:
chore: add ci check for unbuilt updated icons

brian-smith-tcril committed:
chore: rebuild icons

adamstankiewicz committed:
fix: ensure esc key does not close modals that are blocking (#3033)

brian-smith-tcril committed:
feat: add LMS icons (#3029)

Co-authored-by: ihor-romaniuk <ihor.romaniuk@raccoongang.com>

brian-smith-tcril committed:
fix: remove duplicate icons

brian-smith-tcril committed:
chore: make icon verification more robust

arbrandes committed:
Merge pull request #3035 from brian-smith-tcril/fix-extra-icons

fix: remove duplicate icons

arbrandes committed:
Merge pull request #3036 from brian-smith-tcril/more-robust-icon-check

chore: make icon verification more robust

brian-smith-tcril committed:
build: update codecov action to v4 and use token (#3037)

bradenmacdonald committed:
fix: Wrong propTypes for SearchFieldAdvanced.icons (#3026)

BilalQamar95 committed:
chore(deps): updated jest to v29 (#3032)

dependabot[bot] committed:
chore(deps): bump follow-redirects from 1.15.2 to 1.15.6 (#3017)

Bumps [follow-redirects](https://github.com/follow-redirects/follow-redirects) from 1.15.2 to 1.15.6.
- [Release notes](https://github.com/follow-redirects/follow-redirects/releases)
- [Commits](https://github.com/follow-redirects/follow-redirects/compare/v1.15.2...v1.15.6)

---
updated-dependencies:
- dependency-name: follow-redirects
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

brian-smith-tcril committed:
docs: allow-modals in playground (#2987)

dependabot[bot] committed:
chore(deps): bump msgpackr from 1.8.5 to 1.10.1 (#2982)

Bumps [msgpackr](https://github.com/kriszyp/msgpackr) from 1.8.5 to 1.10.1.
- [Release notes](https://github.com/kriszyp/msgpackr/releases)
- [Commits](https://github.com/kriszyp/msgpackr/commits/v1.10.1)

---
updated-dependencies:
- dependency-name: msgpackr
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump axios from 0.27.2 to 0.28.0 (#3007)

Bumps [axios](https://github.com/axios/axios) from 0.27.2 to 0.28.0.
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v0.28.0/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v0.27.2...v0.28.0)

---
updated-dependencies:
- dependency-name: axios
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump express from 4.18.2 to 4.19.2 (#3042)

Bumps [express](https://github.com/expressjs/express) from 4.18.2 to 4.19.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.18.2...4.19.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump es5-ext from 0.10.62 to 0.10.64 (#3012)

Bumps [es5-ext](https://github.com/medikoo/es5-ext) from 0.10.62 to 0.10.64.
- [Release notes](https://github.com/medikoo/es5-ext/releases)
- [Changelog](https://github.com/medikoo/es5-ext/blob/main/CHANGELOG.md)
- [Commits](https://github.com/medikoo/es5-ext/compare/v0.10.62...v0.10.64)

---
updated-dependencies:
- dependency-name: es5-ext
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

edx-requirements-bot committed:
chore: update browserslist DB (#2985)

Co-authored-by: adamstankiewicz <2828721+adamstankiewicz@users.noreply.github.com>

PKulkoRaccoonGang committed:
Merge remote-tracking branch 'origin/master' into Peter_Kulko/alpha-branch-updating

openedx-webhooks commented:
Thanks for the pull request, @PKulkoRaccoonGang! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

PKulkoRaccoonGang committed:
feat: created new design tokens for Chip component

PKulkoRaccoonGang committed:
feat: created new design tokens for ChipCarousel and Pagination components

PKulkoRaccoonGang committed:
refactor: corrected tokens value and removed deprecated Tabs example

PKulkoRaccoonGang commented:
@brian-smith-tcril In this PR the alpha branch has been updated with new changes from the master.
Fixed problems with displaying the Tabs component page and removed the unnecessary overlay `crypt` from `setupTests.js`.
In the next PRs, I plan to check the accuracy of each of the components and add/change existing tokens that we have in the alpha branch.

openedx-webhooks commented:
@PKulkoRaccoonGang 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/3044&entry.1484655318=https://github.com/PKulkoRaccoonGang&entry.752974735=2024-05-07+16:40&entry.1917517419=2024-05-14+15:35&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
